crates/sui-graphql-rpc/src/mutation.rs

```diff
@@ -86,13 +95,40 @@ impl Mutation {
             .map_err(|e| Error::Internal(format!("Unable to execute transaction: {e}")))
             .extend()?;
 
+        let raw_effects: NativeTransactionEffects = bcs::from_bytes(&result.raw_effects)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction effects: {e}")))
+            .extend()?;
+        let sender_signed_data: SenderSignedData = bcs::from_bytes(&result.raw_transaction)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction data: {e}")))
+            .extend()?;
+
+        let events = result
+            .events
+            .ok_or(Error::Internal(
```

Comment:
nit: `ok_or_else` to avoid computing the error in case we don't need it?

```suggestion
            .ok_or_else(|| Error::Internal(
```